packages/editor/src/browser/editor-decorations.ts

```diff
@@ -5,83 +5,9 @@
-import { Range, SetDecorationParams, DeltaDecorationParams, TextEditor } from "./editor";
-import { EditorManager } from "./editor-manager";
-import { DiffUris } from '@theia/core/lib/browser/diff-uris';
-import URI from "@theia/core/lib/common/uri";
+import { Range, TextEditor } from "./editor";
 import { Disposable } from "@theia/core";
 
-@injectable()
-export class EditorDecorationsService {
-
-    protected readonly appliedDecorations = new Map<string, string[]>();
-
-    constructor(
-        @inject(EditorManager) protected readonly editorManager: EditorManager,
-    ) { }
-
-    /**
-     * Applies new decorations of the given kind in an editor opened with the given URI.
-     * Previously applied decoration of the same kind are removed.
-     *
-     * To remove decorations of a kind, client should pass an empty array for new decorations.
-     *
-     * Clients should create qualified kinds for decorations to avoid collision.
-     */
-    async setDecorations(params: SetDecorationParams): Promise<void> {
-        const editor = await this.findEditorByUri(params.uri);
-        if (editor) {
-            const key = `${params.uri}#${params.kind}`;
-            const oldDecorations = this.appliedDecorations.get(key) || [];
-            const appliedDecorations = editor.deltaDecorations(<DeltaDecorationParams & SetDecorationParams>{ oldDecorations, ...params });
-            this.appliedDecorations.set(key, appliedDecorations);
-        }
-    }
-
-    /**
-     * Removes old decorations and applies new decorations in an editor opened with the given URI.
-     *
-     * @returns identifiers of applied decorations, which can be used to remove them in next call,
-     * or `undefined` if no editor could be found with the given uri.
-     */
-    async deltaDecorations(params: DeltaDecorationParams): Promise<string[] | undefined> {
-        const editor = await this.findEditorByUri(params.uri);
-        if (editor) {
-            return editor.deltaDecorations(params);
-        }
-        return undefined;
-    }
-
-    protected async findEditorByUri(uri: string): Promise<TextEditor | undefined> {
-        const editorWidget = await this.editorManager.getByUri(new URI(uri));
-        if (editorWidget) {
-            const editorUri = editorWidget.editor.uri;
-            const openedInEditor = editorUri.toString() === uri;
-            const openedInDiffEditor = DiffUris.isDiffUri(editorUri) && DiffUris.decode(editorUri).some(u => u.toString() === uri);
-            if (openedInEditor || openedInDiffEditor) {
-                return editorWidget.editor;
-            }
-        }
-        return undefined;
-    }
-
-}
-
-export enum OverviewRulerLane {
-    Left = 1,
-    Center = 2,
-    Right = 4,
-    Full = 7
-}
-
-export enum TrackedRangeStickiness {
-    AlwaysGrowsWhenTypingAtEdges = 0,
-    NeverGrowsWhenTypingAtEdges = 1,
-    GrowsOnlyWhenTypingBefore = 2,
-    GrowsOnlyWhenTypingAfter = 3,
-}
-
 export interface EditorDecoration {
```

Comment:
The file name should after the main concept: `editor-decoration.ts`

Reply:
I've updated and splitted up a bit more, see new sub-folder `decorations`.